File: src/main/java/net/dontdrinkandroot/gitki/service/markdown/TableAttributeProvider.java
package net.dontdrinkandroot.gitki.service.markdown;

import org.commonmark.ext.gfm.tables.TableBlock;
import org.commonmark.node.Node;
import org.commonmark.renderer.html.AttributeProvider;

import java.util.Map;

/**
 * @author Philip Washington Sorst <philip@sorst.net>
 */
public class TableAttributeProvider implements AttributeProvider
{
    @Override
    public void setAttributes(Node node, String tagName, Map<String, String> attributes)
    {
        if (node instanceof TableBlock) {
            attributes.put("class", "table");
        }
    }
}

File: src/main/kotlin/net/dontdrinkandroot/gitki/service/markdown/TableAttributeProvider.kt
package net.dontdrinkandroot.gitki.service.markdown

import org.commonmark.ext.gfm.tables.TableBlock
import org.commonmark.node.Node
import org.commonmark.renderer.html.AttributeProvider

class TableAttributeProvider : AttributeProvider {

    override fun setAttributes(node: Node, tagName: String, attributes: MutableMap<String, String>) {
        if (node is TableBlock) attributes["class"] = "table"
    }
}